server/src-lib/Hasura/GraphQL/Execute/RemoteJoins.hs

```diff
@@ -318,28 +309,29 @@ peelOffNestedFields topNames topValue = foldM peel topValue (NE.drop 1 topNames)
             show topValue <> " with " <> show topNames
 
 
--- TODO document all of this
+-- TODO In what sense is this a "batch"?
 data Batch =
   Batch
-    { batchRemoteTopQuery        :: !VQ.RemoteTopField
-    , batchRelFieldPath          :: !RelFieldPath
-    , batchIndices               :: ![ArrayIndex]
+    { batchRelFieldPath          :: !RelFieldPath
```

Comment:
Now that `RemoteTopField` is removed from `Batch` (surprising!), I am not sure what is the point of this data type in the first place. Seems like everything is available in `RemoteRelField` no?

This was supposed to be a batch in the sense it has the query (in the form of VQ.RemoteTopField) which can then make the following query:

```
query {
  hasura_idx_1 : field(..) { .. }
  hasura_idx_2 : field(..) { .. }
  hasura_idx_3 : field(..) { .. }
  ...
}
```